### ISSUE TITLE
Apollo future for desktop clients

### ISSUE THREAD

**Issue by chaosdrop** [ClassicOldSong/Apollo#15]
You're fork of sunshine seems to be a lot more active and going interesting directions, but it seems your only real focus is Android as a client? Is there plans for the desktop versions of moonlight as well, or will this eventually just be an android only server/client when computability with the current Moonlight is eventually broken?

**Comment by ClassicOldSong** [comment]
There will be compatible desktop versions of Artemis in the future. I'm not a superman that can handle every possible client in real time, but at least the desktop version is on my todo list.

**Comment by chaosdrop** [comment]
Sounds fair. I just wanted to make sure there was a future for desktop before I switched over since it kind of sounded like it could be Android only over the long run just reading the github page.

If you ever do make a desktop version of Artemis the major feature I think is missing from Moonlight/sunshine (at least from a VNC / remote desktop replacement standpoint) is the option to require a password to connect to different severs. Kind of bothers me that if someone is using my computer they can open moonlight and have access to all the computers I have ever connected to without needing a password since a lot of those computers are other peoples.

**Comment by tkazbekov2** [comment]
@chaosdrop would the pass be set on the host computer then? What about permission system, like in parsec?

**Comment by chaosdrop** [comment]
> @chaosdrop would the pass be set on the host computer then? What about permission system, like in parsec?

Parsec mostly worked for me before but I moved to Shinshine/Moonlight but still wasn't a big fan of it's security system [but loved the performance over VNC]. I like how VNC clients work where you can just have a password set for every computer. 

So yes, the password would be set on each host computer, and then each client connecting to that computer would need to provide the password to connect. In an ideal setup just like tightVNC you'd have different passwords for Viewing and interacting.

**Comment by tiamed** [comment]
> Currently I'm only doing additions, which doesn't break compatibility with other moonlight clients. Feature wise I don't think there're things that I need to modify deliberately to break compatibility with current Moonlight clients, and Artemis is still compatible with other Sunshine hosts. Just the newly added features are coupled to provide a better experience.

On the linux client I launch `Desktop` and got a blackscreen, is it related to the resolution detection from the vritual display initialization? For the android client it works fine, and starting the session on android and continue it on linux is a workaround now.

**Comment by ClassicOldSong** [comment]
It looks like an encoder/decoder issue.

After recent Nvidia graphics driver upgrade my Android will also get an initial black screen but move something on the virtual display like shaking the mouse may get it working again. There was a workaround on the initial black screen issue but it doesn't seem working now.

**Comment by ZhaoZelin2000** [comment]
Does the desktop client have a schedule? It seems that clipboard synchronization does not work for Moonlight PC

**Comment by ClassicOldSong** [comment]
Indeed it's not working. Clipboard sync is Apollo's specific feature, OG moonlight will most likely not supporting it unless it decides to change the main compatible target to Apollo.

I'm too busy to make another fork recently, so sadly that you still have to wait for a while...

**Comment by francois-pasquier** [comment]
What about adding the virtual display, setting resolution etc. / removing it when connection/disconnecting?

It would require the Apollo desktop version right? It wouldn't work with moonlight would it?

**Comment by ClassicOldSong** [comment]
Virtual Display works with ANY moonlight client.

**Comment by ClassicOldSong** [comment]
For iOS client you can use Moonlight-ZWM, it's already on AppStore.

Any other moonlight clients can connect to Apollo just fine.

**Comment by MianReplicate** [comment]
Well I know that they can connect just fine, I use them at the moment but I thought you were going to eventually drop support for Moonlight clients for Sunshine? Unless that specific moonlight client you mentioned specifically supports Apollo

**Comment by ClassicOldSong** [comment]
I will not drop support deliberately, it'll only happen when current protocol fundamentally cannot handle future needs. So it's not going to happen anytime soon.

**Comment by ProTechnomancy** [comment]
Can't apk's run on win11 these days anyway?

**Comment by ClassicOldSong** [comment]
You can compile it on Linux right now, but there's no virtual display support for Linux.

In the future there will be out of the box support for Linux, but not in the near future as Linux is really fragmented and really hard to work with.

**Comment by rogerogden** [comment]
@ClassicOldSong Thank you for Apollo, it looks great. I'm wondering if it's currently possible to compile Apollo on macOS, which, I believe, does have virtual display support. I'm currently using a beta version of Sunshine on macOS, but would be interested in moving to Apollo if possible.

Thanks!

**Comment by ClassicOldSong** [comment]
I think it's possible to compile now(haven't tried myself), but without virtual display feature.

Permissions and other improvements can work.

**Comment by ClassicOldSong** [comment]
Yeah, there will, this is not something really difficult to implement.

**Comment by Xarishark** [comment]
+1 for the flatpak build instead of a bunch of distro depended package systems.
